frontend/src/scenes/instance/DeadLetterQueue/deadLetterQueueLogic.ts

```diff
@@ -26,22 +33,48 @@ export const deadLetterQueueLogic = kea<deadLetterQueueLogicType>({
                 setActiveTab: (_, { tabKey }) => tabKey,
             },
         ],
+        rowsPerMetric: [
+            {} as Record<string, any[]>,
+            {
+                addRowsToMetric: (state, { key, rows }) => {
+                    return { ...state, [key]: [...(state[key] || []), ...rows] }
+                },
+                setRowsPerMetricKey: (_, { map }) => map,
+            },
+        ],
     }),
 
-    loaders: () => ({
+    loaders: ({ actions }) => ({
         deadLetterQueueMetrics: [
-            [] as SystemStatusRow[],
+            [] as DeadLetterQueueMetricRow[],
             {
                 loadDeadLetterQueueMetrics: async () => {
                     if (!userLogic.values.user?.is_staff) {
                         return []
                     }
-                    return (await api.get('api/dead_letter_queue')).results?.overview
+                    const metrics = (await api.get('api/dead_letter_queue')).results
+
+                    const rowsPerMetric = {}
+                    for (const metric of metrics.filter((m: DeadLetterQueueMetricRow) => !!m.subrows)) {
+                        rowsPerMetric[metric.key] = metric.subrows.rows
+                    }
+                    actions.setRowsPerMetricKey(rowsPerMetric)
```

Comment:
All of this can be removed and added into the reducer for `rowsPerMetrics`, for the `loadDeadLetterQueueMetricsSuccess` action. 

A loader should ideally just be responsible for getting the raw data, not doing any additional fancy processing and piping of it.

Reply:
> for the `loadDeadLetterQueueMetricsSuccess` action.


ah nice!